projects/components/src/file-upload/file-upload-validators.ts

```diff
@@ -0,0 +1,133 @@
+import { AbstractControl } from '@angular/forms';
+import { sumBy } from 'lodash-es';
+import { FileItem } from '../file-display/file-item';
+import { UploadFileType } from './upload-file-type';
+
+/**
+ * @description
+ * Provides a set of validators specifically for file upload component that can be used by form controls.
+ */
+// tslint:disable:no-null-keyword
+// tslint:disable:no-unnecessary-class
+export class FileUploadValidators {
+  /**
```

Comment:
Lots of comments!!!

Reply:
Yeah, basically I have used the same approach as angular to add new custom validators. These comments will be seen in VSCode hover for these functions.